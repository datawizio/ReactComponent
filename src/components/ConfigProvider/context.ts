import { createContext } from "react";

export interface ConfigContextValue {
  translate: (transKey: string) => string;
}

const ConfigContext = createContext<ConfigContextValue>({
  translate: transKey => transKey
});

export default ConfigContext;
