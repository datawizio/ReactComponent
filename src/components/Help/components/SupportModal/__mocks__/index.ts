export const file = {
  "uid": "rc-upload-1624447080939-12",
  "lastModified": 1624346309471,
  "lastModifiedDate": "2021-06-22T07:18:29.471Z",
  "name": "main_logo.png",
  "size": 151625,
  "type": "image/png",
  "percent": 100,
  "originFileObj": {
    "uid": "rc-upload-1624447080939-12"
  },
  "status": "done",
  "response": {
    "upload": {
      "token": "Gw6zeacBaxwV6hmB2i5j1yIaD",
      "expires_at": "2021-06-26T12:32:08Z",
      "attachments": [
        {
          "url":
            "https://datawiz.zendesk.com/api/v2/attachments/371822025780.json",
          "id": 371822025780,
          "file_name": "main_logo.png",
          "content_url":
            "https://datawiz.zendesk.com/attachments/token/EV6As3zABAHRrPtIf7CrmQYHd/?name=main_logo.png",
          "mapped_content_url":
            "https://datawiz.zendesk.com/attachments/token/EV6As3zABAHRrPtIf7CrmQYHd/?name=main_logo.png",
          "content_type": "image/png",
          "size": 151625,
          "width": 669,
          "height": 670,
          "inline": false,
          "deleted": false,
          "thumbnails": [
            {
              "url":
                "https://datawiz.zendesk.com/api/v2/attachments/371822025900.json",
              "id": 371822025900,
              "file_name": "main_logo_thumb.png",
              "content_url":
                "https://datawiz.zendesk.com/attachments/token/IcekaptdzghWveJ401aETDZuh/?name=main_logo_thumb.png",
              "mapped_content_url":
                "https://datawiz.zendesk.com/attachments/token/IcekaptdzghWveJ401aETDZuh/?name=main_logo_thumb.png",
              "content_type": "image/png",
              "size": 2420,
              "width": 80,
              "height": 80,
              "inline": false,
              "deleted": false
            }
          ]
        }
      ],
      "attachment": {
        "url":
          "https://datawiz.zendesk.com/api/v2/attachments/371822025780.json",
        "id": 371822025780,
        "file_name": "main_logo.png",
        "content_url":
          "https://datawiz.zendesk.com/attachments/token/EV6As3zABAHRrPtIf7CrmQYHd/?name=main_logo.png",
        "mapped_content_url":
          "https://datawiz.zendesk.com/attachments/token/EV6As3zABAHRrPtIf7CrmQYHd/?name=main_logo.png",
        "content_type": "image/png",
        "size": 151625,
        "width": 669,
        "height": 670,
        "inline": false,
        "deleted": false,
        "thumbnails": [
          {
            "url":
              "https://datawiz.zendesk.com/api/v2/attachments/371822025900.json",
            "id": 371822025900,
            "file_name": "main_logo_thumb.png",
            "content_url":
              "https://datawiz.zendesk.com/attachments/token/IcekaptdzghWveJ401aETDZuh/?name=main_logo_thumb.png",
            "mapped_content_url":
              "https://datawiz.zendesk.com/attachments/token/IcekaptdzghWveJ401aETDZuh/?name=main_logo_thumb.png",
            "content_type": "image/png",
            "size": 2420,
            "width": 80,
            "height": 80,
            "inline": false,
            "deleted": false
          }
        ]
      }
    }
  },
  "xhr": {}
};
