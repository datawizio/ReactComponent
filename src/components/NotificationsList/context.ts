import { createContext, Dispatch } from "react";
import { Action, IListItem, ListState } from "./types";

export const NotificationsListContext = createContext<{
  state: ListState<IListItem> | null;
  dispatch: Dispatch<Action<IListItem>> | null;
}>({ state: null, dispatch: null });

NotificationsListContext.displayName = "NotificationsListContext";
