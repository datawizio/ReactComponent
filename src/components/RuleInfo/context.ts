import { createContext } from "react";
import { IRuleInfoContext } from "./types";

export const RuleInfoContext = createContext<IRuleInfoContext>(null);

RuleInfoContext.displayName = "RuleInfoContext";
