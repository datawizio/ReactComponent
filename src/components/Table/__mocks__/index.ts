import { Action, BodyCellType, IColumn, TableState } from "../types";
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";
import { genColumns, genDataSource } from "../../../utils/data/dataGenerators";

const key = Math.random().toString(36).substring(7);

// defineCellType
export const validColumn: IColumn = {
  dtype: "dtype",
  dataIndex: "dataIndex",
  resizable: false
};

export const invalidColumn: IColumn = {
  dataIndex: "dataIndex",
  resizable: false
};

export const cell: BodyCellType = {
  dtype: "dtype",
  key: "key"
};

// filterByColumns
export const columnsMap: TableState["columnsMap"] = {
  col1: {
    dtype: "dtype",
    dataIndex: "1",
    resizable: true
  },
  col2: {
    dtype: "dtype2",
    dataIndex: "2",
    resizable: true
  },
  col3: {
    dtype: "dtype3",
    dataIndex: "3",
    resizable: true
  },
  col4: {
    dtype: "dtype4",
    dataIndex: "4",
    resizable: true
  }
};

export const obj: { [columnKey: string]: any } = {
  col1: {},
  col2: "any",
  col5: "string",
  col6: "number"
};

export const expectedOutput_filterByColumns = {
  col1: {},
  col2: "any"
};

// swapColumns
const generateCollumn = (isValid: boolean = true): IColumn => {
  const column = isValid ? { ...validColumn } : { ...invalidColumn };
  return {
    key,
    children: [
      {
        key,
        ...validColumn
      }
    ],
    ...column
  };
};

export const columnsMock: Array<IColumn> = [
  ...[...new Array(3)].map(() => generateCollumn(true)),
  ...[...new Array(2)].map(() => generateCollumn(false))
];

//Column

export const model = {
  key: "key",
  resizable: false,
  fixed: true,
  title: "title"
};

//Table

const columnsCount = number("columns count", 3);
const dataCount = number("data count", 3);
const treeColumns = boolean("treeColumns", true);
const treeData = boolean("treeData", true);
const toolBar = boolean("toolbar", true);
export const sortable = boolean("sortable", true);

export const columns = genColumns(columnsCount, treeColumns);

export let dataSource = genDataSource(dataCount, columns, ["string"], treeData);

export const getStaticDataSource = () => {
  return [
    {
      key: "0-0",
      "alias10-20279": "Garden",
      "alias20-19186": 93415,
      "alias20-21247": 69154,
      children: [
        {
          key: "0-0",
          "alias10-20279": "Garden",
          "alias20-19186": 93415,
          "alias20-21247": 69154
        }
      ]
    },
    {
      key: "1-0",
      "alias10-20279": "Prairie",
      "alias20-19186": 66323,
      "alias20-21247": 66325
    },
    {
      key: "2-0",
      "alias10-20279": "Generic",
      "alias20-19186": 1440,
      "alias20-21247": 73323
    }
  ];
};
// if (columns[0]) {
//   columns[0].fixed = "left";
//   columns[0].filters = [
//     // @ts-ignore
//     ...new Set(dataSource.map(item => item[columns[0].dataIndex]))
//   ].map(item => ({
//     text: item,
//     value: item
//   }));
// }

// Handlers

export const dTypeConfig = {
  boolean: {},
  number: {},
  string: {}
};

//CellData
export const getStaticColumn = (levelOfNesting: number = 1) => {
  const hasChilds = levelOfNesting > 1;

  return {
    dataIndex: "alias20-89663",
    dType: "render",
    key: "0-47391-2",
    onHeaderCell: jest.fn(),
    render: jest.fn(),
    sorter: true,
    title: "Alias",
    ...(hasChilds && { children: getStaticColumn(levelOfNesting - 1) })
  };
};

export function getStaticRow(levelOfNesting: number = 1) {
  const hasChilds = levelOfNesting > 1;

  return {
    address: "city",
    age: 10,
    ...(hasChilds && { children: getStaticRow(levelOfNesting - 1) }),
    key: levelOfNesting,
    name: "name"
  };
}

//Column
export const getStatickModel = (levelOfNesting: number = 1) => {
  const hasChilds = levelOfNesting > 1;

  return {
    dataIndex: "aut12-71229",
    key: "",
    onHeaderCell: () => {},
    render: () => {},
    resizable: true,
    ...(hasChilds && { children: getStatickModel(levelOfNesting - 1) }),
    title: "title"
  };
};

export const getDTypeConfig = () => ({
  sorted: () => true,
  toString: () => "toString",
  search: () => true,
  filter: () => true,
  render: () => true
});

// Table reducer
export const updateDataSourceAC = (payload): Action => ({
  type: "updateDataSource",
  payload: payload
});

export const updateColumnsAC = (payload): Action => ({
  type: "updateColumns",
  payload: payload || []
});

export const visibleColumnsKeysAC = (payload): Action => ({
  type: "visibleColumnsKeys",
  payload
});

export const paginateAC = (payload): Action => ({
  type: "paginate",
  payload
});

export const resetPaginationAC = (): Action => ({
  type: "resetPagination"
});

export const searchAC = (payload): Action => ({
  type: "search",
  payload
});

export const sorthAC = (payload): Action => ({
  type: "sort",
  payload: [].concat(payload)
});

export const filterAC = (payload): Action => ({
  type: "filter",
  payload
});

export const addLoadingRowAC = (payload): Action => ({
  type: "addLoadingRow",
  payload
});

export const setRowChildrenAC = (payload): Action => ({
  type: "setRowChildren",
  payload
});

export const swapColumnsAC = (payload): Action => ({
  type: "swapColumns",
  payload
});

export const expandRowAC = (payload): Action => ({
  type: "expandRow",
  payload
});

export const collapseRowAC = (payload): Action => ({
  type: "collapseRow",
  payload
});

export const loadingAC = (payload): Action => ({
  type: "loading",
  payload
});

export const recoveryStateAC = (payload): Action => ({
  type: "recoveryState",
  payload
});

export const updateRowAC = (payload): Action => ({
  type: "updateRow",
  payload
});

export const updateAC = (payload): Action => ({
  type: "update",
  payload
});
