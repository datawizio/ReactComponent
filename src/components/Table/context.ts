import React, { Dispatch } from "react";
import { TableProps, TableState, Action } from "./types";

export const TableContext = React.createContext<{
  tableState: TableState;
  tableProps: TableProps;
  baseTableState: TableState;
  dispatch: Dispatch<Action>;
}>(null);

TableContext.displayName = "TableContext";
