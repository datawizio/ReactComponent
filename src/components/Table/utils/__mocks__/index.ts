import { BodyCellType, IColumn } from "../../types";

export const validColumn: IColumn = {
  dtype: "dtype",
  dataIndex: "dataIndex",
  resizable: false
};

export const cell: BodyCellType = {
  dtype: "dtype",
  key: "key"
};

export const invalidColumn: IColumn = {
  dataIndex: "dataIndex",
  resizable: false
};
