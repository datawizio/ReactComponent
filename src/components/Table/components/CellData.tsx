import * as React from "react";
import { IRow, IColumn } from "../types";
import { TableContext } from "../context";
import { useMemo, useContext } from "react";
import { defineCellType } from "../utils/utils";

export interface CellDataProps<T = any> {
  value: T;
  row: IRow;
  column: IColumn;

  yIndex: number;
  xIndex: number;
  columnLevel: number;
}

const CellData: React.FC<CellDataProps> = React.memo(props => {
  const { value, row, column, xIndex, yIndex, columnLevel } = props;
  const [{ dTypesConfig }, , { cellRenderProps, rowPrefix }] = useContext(
    TableContext
  );

  const typeConfig = useMemo(() => {
    return dTypesConfig[defineCellType(value, column)];
  }, [dTypesConfig, value, column]);

  const cellDataRender = useMemo(() => {
    return typeConfig && typeConfig.render
      ? typeConfig.render(value, row, column, xIndex, cellRenderProps)
      : value;
  }, [typeConfig, value, xIndex, cellRenderProps, column, row]);

  const rowPrefixRender = useMemo(() => {
    return (
      rowPrefix &&
      yIndex === 0 &&
      columnLevel === 1 &&
      rowPrefix(value, row, column, xIndex)
    );
  }, [rowPrefix, value, xIndex, column, row, columnLevel, yIndex]);

  return (
    <>
      <span className="row-prefix">{rowPrefixRender}</span>
      {cellDataRender}
    </>
  );
});

export default CellData;
