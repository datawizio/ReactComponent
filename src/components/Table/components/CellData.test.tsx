import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import CellData from "./CellData";
import { getStaticRow, getDTypeConfig, getStaticColumn } from "../__mocks__";
import { TableContext } from "../context";

const context = {
  tableState: { dTypesConfig: getDTypeConfig() },
  tableProps: {
    cellRenderProps: undefined,
    rowPrefix: jest.fn()
  }
};

const componentWithContext = props => {
  return (
    //@ts-ignore
    <TableContext.Provider value={context}>
      <CellData {...props} />
    </TableContext.Provider>
  );
};

const setUp = (props?) => mount(componentWithContext(props));

describe("CellData component", () => {
  let component;
  const column = getStaticColumn();
  const celllDataProps = {
    value: "35",
    row: getStaticRow(2),
    column,
    xIndex: 1,
    yIndex: 2,
    columnLevel: 1
  };
  beforeEach(() => {
    component = setUp(celllDataProps);
  });

  it("rendered CellData corectly", () => {
    expect(component).toMatchSnapshot();
  });

  it("CellData has cotrect classes ", () => {
    expect(component.find(".dw-table__cell-data").length).toBe(1);
  });
});
