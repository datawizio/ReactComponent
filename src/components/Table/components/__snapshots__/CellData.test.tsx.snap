// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellData component rendered CellData corectly 1`] = `
<Memo()
  column={
    Object {
      "dType": "render",
      "dataIndex": "alias20-89663",
      "key": "0-47391-2",
      "onHeaderCell": [MockFunction],
      "render": [MockFunction],
      "sorter": true,
      "title": "Alias",
    }
  }
  columnLevel={1}
  row={
    Object {
      "address": "city",
      "age": 10,
      "children": Object {
        "address": "city",
        "age": 10,
        "key": 1,
        "name": "name",
      },
      "key": 2,
      "name": "name",
    }
  }
  value="35"
  xIndex={1}
  yIndex={2}
>
  <div
    className="dw-table__cell-data"
  >
    35
  </div>
</Memo()>
`;
